src/qibo/models/variational.py

```diff
@@ -67,24 +66,27 @@ def minimize(self, initial_state, method='Powell', jac=None, hess=None,
         """
         def _loss(params, circuit, hamiltonian):
             circuit.set_parameters(params)
-            final_state = circuit()
+            final_state = circuit().state()
             return hamiltonian.expectation(final_state)
 
         if compile:
-            if K.is_custom:
+            #TODO: introduce is_custom attribute or maybe just compare with qibojit
+            if not self.hamiltonian.backend == 'qibojit':
```

Comment:
I think this causes some issues in the tensorflow tests. I don't think we need the `is_custom` attribute anymore since qibotf is removed, instead I would do one of:

1. Change this to
```suggestion
            if not self.hamiltonian.backend == 'tensorflow':
```
since tensorflow is the only backend that supports compilation.

2. Remove this check completely. If `backend.compile` is used with numpy or qibojit it will still work but will have no effect.

Reply:
I've ended up remove this check.